demos/utils/cloud.py

```diff
@@ -0,0 +1,158 @@
+
+# Copyright 2023 The ClusterLink Authors.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import subprocess as sp
+import time
+import os
+from demos.utils.k8s import  cleanCluster, waitPod, getNodeIP
+from demos.utils.common import runcmd , createPeer, applyPeer, printHeader
+
+# cluster class represents a cloud cluster for deploying the ClusterLink gateway. 
+class cluster:
+    def __init__(self, name, zone,platform,machineType="small"):
+        self.name        = name
+        self.zone        = zone
+        self.platform    = platform
+        self.machineType = machineType
+        self.port        = 443
+        self.ip          = ""
+        self.nodeIP      = "" # the IP of the worker node
+    
+    # createCluster creates a K8s cluster on cloud platform.
+    def createCluster(self, runBg):
```

Comment:
Can we have a method to setCluster parameters, if the cluster is already present and someone wants to test deployment, without having to recreate cluster?

Reply:
Yes I split it to createCluster and startCluster